File: secretspec-swift/Examples/QuickStart.swift
import SecretSpec

func quickStart() throws {
    let resolved = try SecretSpec.builder()
        .withProvider("keyring://")
        .withProfile("production")
        .withReason("boot web app")
        .load()
    defer { try? resolved.close() }

    print(resolved.provider, resolved.profile)
    print(resolved.secrets["DATABASE_URL"]?.get() ?? "")
    try resolved.setAsEnvironment()
}

File: secretspec-jvm/examples/quick_start/QuickStart.java
import org.cachix.secretspec.SecretSpec;

public class QuickStart {

    public static void main() {
        try(var resolved = SecretSpec.builder()
            .withProvider("keyring://")
            .withProfile("production")
            .withReason("boot web app")
            .load()
        ) {
            System.out.println(resolved.getProvider() + " (" + resolved.getProfile() + ")");
            var secrets = resolved.getSecrets();
            System.out.println(secrets.get("DATABASE_URL").get());
            resolved.setAsSystemProperties();
        }    
    }
}
